src/mui/input/SelectArrayInput.js

```diff
@@ -71,6 +84,26 @@ export class SelectArrayInput extends Component {
         }
     };
 
+    existsInChoices = (obj) => {
+        const { optionText } = this.props;
+        const text = typeof obj === 'string' ? obj : obj.text;
+        return this.props.choices.find(choice => choice[optionText] === text);
+    };
+
+    handleCreatedRecord = (createdRecord) => {
+        const { optionText, optionValue } = this.props;
+        const values = [...this.state.values, {
+            text: createdRecord[optionText],
+            value: createdRecord[optionValue],
+        }];
+        this.setState({ values });
+        this.handleChange(values);
+
+        setTimeout(() => {
```

Comment:
You'll need to add a comment to explain the `setTimeout`, where the `this.chipInput` comes from (the `ref` in `render()`), and why you need to do direct DOM manipulation instead of usual React.